kotlin/workflow-core/src/main/java/com/squareup/workflow/Workflow.kt

```diff
@@ -73,4 +75,63 @@ interface Workflow<in InputT : Any, out OutputT : Any, out RenderingT : Any> {
    * the common API required for [WorkflowContext.compose] to do its work.
    */
   fun asStatefulWorkflow(): StatefulWorkflow<InputT, *, OutputT, RenderingT>
+
+  companion object {
+
+    /**
+     * A convenience function to implement [stateless] by just passing the
+     * [compose][stateless.compose] function as a lambda.
+     *
+     * Note that while a stateless workflow doesn't have any _internal_ state of its own, it may use
+     * [input][InputT] received from its parent, and it may compose child workflows that do have their own
+     * internal state.
+     */
+    fun <InputT : Any, OutputT : Any, RenderingT : Any> stateless(
+      compose: (
+        input: InputT,
+        context: WorkflowContext<Nothing, OutputT>
+      ) -> RenderingT
+    ): StatelessWorkflow<InputT, OutputT, RenderingT> =
+      object : StatelessWorkflow<InputT, OutputT, RenderingT>() {
+        override fun compose(
+          input: InputT,
+          context: WorkflowContext<Nothing, OutputT>
+        ): RenderingT = compose.invoke(input, context)
+      }
+
+    /**
+     * A version of [stateless] that doesn't have any input data either.
+     */
+    fun <OutputT : Any, RenderingT : Any> stateless(
+      compose: (context: WorkflowContext<Nothing, OutputT>) -> RenderingT
+    ): StatelessWorkflow<Unit, OutputT, RenderingT> =
+      stateless { _: Unit, context: WorkflowContext<Nothing, OutputT> ->
+        compose(context)
+      }
+
+    /**
+     * A version of [stateless] that does nothing but compose a given [rendering].
+     * Handy for testing.
+     */
+    fun <OutputT : Any, RenderingT : Any> stateless(
+      rendering: RenderingT
+    ): StatelessWorkflow<Unit, OutputT, RenderingT> =
+      stateless { _: Unit, _: WorkflowContext<Nothing, OutputT> -> rendering }
+  }
+}
+
+/**
+ * Uses the given [function][transform] to transform a [Workflow] that
+ * renders [FromRenderingT] to one renders [ToRenderingT],
+ */
+// Intellij refuses to format this parameter list correctly because of the weird line break,
+// and detekt will complain about it.
+// @formatter:off
+fun <InputT : Any, OutputT : Any, FromRenderingT : Any, ToRenderingT : Any>
+    Workflow<InputT, OutputT, FromRenderingT>.mapRendering(
```

Comment:
Wait where did this live before? I don't see it being deleted from anywhere in this PR.

Reply:
Only method in Workflows.kt, the deleted file is collapsed in the diff.

Reply:
Ah, derp

Reply:
Anyway much prefer this method living in this file, thanks!

Reply:
Anyway much prefer this method living in this file, thanks!